apiary.apib

```diff
@@ -954,11 +954,568 @@ Outputs the list of peers that were removed.
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        - Key: "Datastore.Path" (string)
+        - Value: "kitten" (string)
+
+
+    + Body
+
+        ```
+        {
+          "Key": "Datastore.Path",
+          "Value": "kitten"
+        }
+        ```
+
++ Request With POST and JSON Flag With Invalid JSON Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/config?arg=Datastore.Path&arg=kitten&json"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/config?arg=Datastore.Path&arg=kitten&json"
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Tue, 02 Feb 2016 22:26:11 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "failed to unmarshal json. invalid character 'k' looking for beginning of value"
+        - Code: 0
+
+
+    + Body
+
+        ```
+        {
+          "Message": "failed to unmarshal json. invalid character 'k' looking for beginning of value",
+          "Code": 0
+        }
+        ```
+
++ Request With POST and JSON Flag With Valid JSON Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/config?arg=Datastore.Path&arg=\"kitten\"&json"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/config?arg=Datastore.Path&arg=\"kitten\"&json"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Tue, 02 Feb 2016 22:28:37 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        - Key: "Datastore.Path" (string)
+        - Value: "kitten" (string)
+
+
+    + Body
+
+        ```
+        {
+          "Key": "Datastore.Path",
+          "Value": "kitten"
+        }
+        ```
+
+## edit [GET /config/edit]
+Opens the config file for editing in $EDITOR
+
++ Request
+
+    ### curl
+
+        curl -i http://localhost:5001/api/v0/config/edit
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/config/edit
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Tue, 02 Feb 2016 17:54:56 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes
+
+    + Body
+
+        ```
+        ```
+
+## replace [GET /config/replace{?arg}]
+Replaces the config with <file>
+
+Make sure to back up the config file first if neccessary, this operation
+can't be undone.
+
+#### curl
+
+    curl -i -F "file=@test" http://localhost:5001/api/v0/config/replace
```

Comment:
I'm confident that if -F is passed to curl, it executes a POST request and not a GET (above says it is a GET)